### PR TITLE
fix(cli): throw an error if a stack contains a whitespace

### LINKED ISSUE #1741: cdktf synth doesn't generate lambda assets archive when TerraformStack name contains whitespaces
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
When `TerraformStack` name parameter contains whitespaces, the assets archive is not created.

Sample code:
<pre>
new LambdaStack(app, '<b>lambda hello world</b>', {
  path: "../lambda-hello-world/dist",
  handler: "index.handler",
  runtime: "nodejs14.x",
  stageName: "hello-world",
  version: "v0.0.2"
});
</pre>

### cdktf & Language Versions

`"cdktf": "0.10.2"`
`"typescript": "^4.5.2"`

### Affected Resource(s)

`TerraformAsset`

### Debug Output
No errors, process finishes without error logs

### Expected Behavior

When using `cdktf synth` with lambda, the `cdktf.out/stacks/{lambda-stack-name}/assets` should have archived lambda source code.

### Actual Behavior

`cdktf.out/stacks/{lambda-stack-name}/assets` folder doesn't contain archived lambda code

### Steps to Reproduce

Clone the repo https://github.com/hashicorp/learn-cdktf-assets-stacks-lambda

Replace the lambda stack name `lambda-hello-world` with `lambda hello world`

run `cdktf synth`

check the output folder `cdktf.out/stacks/{lambda-stack-name}/assets`

### Important Factoids

No

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Closes #1741

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## 0.11.0 (unreleased)
+
+**Breaking Changes**
+
+### Stack ids can no longer contain a whitespace
```
Comment (ansgarm) [inline CHANGELOG.md:5]:
```suggestion
### Stack ids can no longer contain whitespaces
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+
+A `TerraformStack` may no longer contain whitespace characters, since we rely on paths being whitespace free. If you have a stack with an ID containing a whitespace, please replace it with a hyphen.
```
Comment (ansgarm) [inline CHANGELOG.md:7]: The header states "ids" and here it is "ID", let's use only "id"?

File: packages/cdktf/lib/terraform-stack.ts [hunk 3]
```diff
@@ -45,6 +45,14 @@ function throwIfIdIsGlobCharacter(str: string): void {
   });
 }
 
+function throwIfIdContainsWhitespace(str: string): void {
+  if (str.includes(" ")) {
```
Comment (ansgarm) [inline packages/cdktf/lib/terraform-stack.ts:49]: Should we extend this to any whitespace character while were at it?
Maybe use a regex with `\s`?

File: packages/cdktf/lib/terraform-stack.ts [hunk 4]
```diff
@@ -45,6 +45,14 @@ function throwIfIdIsGlobCharacter(str: string): void {
...
+    throw new Error(
+      `Can not create Terraform stack with id "${str}". It contains a whitespace.`
```
Comment (ansgarm) [inline packages/cdktf/lib/terraform-stack.ts:51]:
```suggestion
      `Can not create TerraformStack with id "${str}". It contains a whitespace.`
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+## 0.11.0 (unreleased)
+
+**Breaking Changes**
+
+### Stack ids can no longer contain whitespaces
+
+A `TerraformStack` may no longer contain whitespace characters, since we rely on paths being whitespace free. If you have a stack with an id containing a whitespace, please replace it with a hyphen.
```
Comment (ansgarm) [inline CHANGELOG.md:7]: Sorry for missing this in my first review, but renaming a stack could change the name of the default local backend state file, right? I feel like we should warn users that they might lose state (maybe link them to some docs about renaming stacks – if we have such 😅)

Comment (DanielMSchmidt) [reply CHANGELOG.md:7]: No good point, I entirely missed that

**Review by ansgarm — APPROVED** [review]

**Pull request merged**